Packet++/src/S7CommLayer.cpp

```diff
@@ -0,0 +1,117 @@
+#include "EndianPortable.h"
+
+#include "S7CommLayer.h"
+#include "TcpLayer.h"
+#include <iostream>
+
+#include <cstring>
+#include <sstream>
+
+namespace pcpp
+{
+
+	S7CommLayer::S7CommLayer(uint8_t msgType, uint16_t pduRef, uint16_t paramLength, uint16_t dataLength,
+							 uint8_t errorClass, uint8_t errorCode)
+	{
+		size_t headerLen;
+		if (msgType == 0x03)
+		{
+			headerLen = sizeof(s7comm_ack_data_hdr);
+			m_DataLen = headerLen;
+			m_Data = new uint8_t[headerLen];
+			memset(m_Data, 0, headerLen);
+			s7comm_ack_data_hdr *ack_d = (s7comm_ack_data_hdr *)m_Data;
+			ack_d->protocolId = 0x32;
+			ack_d->msgType = msgType;
+			ack_d->reserved = 0x0000;
+			ack_d->pduRef = htobe16(pduRef);
+			ack_d->paramLength = htobe16(paramLength);
+			ack_d->dataLength = htobe16(dataLength);
+			ack_d->errorClass = errorClass;
+			ack_d->errorCode = errorCode;
+		}
+		else
+		{
+			headerLen = sizeof(s7commhdr);
+			m_DataLen = headerLen;
+			m_Data = new uint8_t[headerLen];
```

Comment:
Ditto: the data len is not correct I think, because the actual length should be: `sizeof(s7commhdr) + paramLength + dataLength`. We should also set `m_Data` to this size

Reply:
@wivien19 you didn't fix this one